// __BEGIN_LICENSE__
// Copyright (C) 2006-2010 United States Government as represented by
// the Administrator of the National Aeronautics and Space Administration.
// All Rights Reserved.
// __END_LICENSE__


package vw.platefile;

message BBoxContainer {
  required float origin_x = 1;
  required float origin_y = 2;
  required float width    = 3;
  required float height   = 4;
}

message WMSTileRequest {
  required int32 platefile_id   = 1;
  required BBoxContainer lonlat = 2;
  required BBoxContainer pixels = 3;
}

message WMSTileResponse {
  required string filename = 1;
}

service WMSService {
  //rpc ListRequest (WMSCapRequest) returns (WMSCapResponse);
  rpc GetTile (WMSTileRequest) returns (WMSTileResponse);
}
